### PR TITLE
Add missing typecheck command

### PR DESCRIPTION
**Pull request by itsdouges**
This pull request adds a typecheck command and adds it to the CI run. It also fixes a ton of violations. One thing I haven't reconciled yet is how to typecheck the project references with the same command, any ideas let me know! Perhaps we just exclude packages and assume they're typechecked when generating the types?...

### CODE REVIEW

File: packages/tsconfig.options.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
     "resolveJsonModule": true,
     "sourceMap": true,
     "strict": true,
+    "skipLibCheck": true,
```
Comment (pancaspe87) [inline packages/tsconfig.options.json:19]: What does this do?

File: stories/css-prop-static-object.tsx [hunk 2]
```diff
@@ -23,7 +23,6 @@ export const ObjectLiteralSpreadFromFunc = (): JSX.Element => {
       css={{
         display: 'flex',
         fontSize: '50px',
-        color: 'blue',
```
Comment (pancaspe87) [inline stories/css-prop-static-object.tsx:26]: Why are we removing this?

File: packages/tsconfig.options.json (hunk 1 above)
Comment (itsdouges) [reply packages/tsconfig.options.json:19]: Uh was just testing, I should remove it.

File: stories/css-prop-static-object.tsx (hunk 2 above)
Comment (itsdouges) [reply stories/css-prop-static-object.tsx:26]: It's a type error :-) "color is defined multiple times and this won't be used"

**Comment by dddlr** [comment]
Ooh thanks for the fix

LGTM assuming that `skipLibCheck` is removed and you get green builds

**Pull request closed without merging**